### ISSUE TITLE
`IllegalStateException` in `ContentSource.demand()`

### ISSUE THREAD

**Issue by olenagerasimova** [jetty-project/jetty-reactive-httpclient#323]
Jetty 12.0.3, jetty-reactive-httpclient 4.0.1, java 21

Sometimes, if connection is slow, I'm getting the following error:

```
java.lang.IllegalStateException
	at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer$ContentSource.demand(ResponseListeners.java:665)
	at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.read(ResponseListenerProcessor.java:197)
	at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.run(ResponseListenerProcessor.java:181)
	at org.eclipse.jetty.reactive.client.internal.QueuedSinglePublisher.tryProduce(QueuedSinglePublisher.java:70)
	at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.onRequest(ResponseListenerProcessor.java:174)
	at org.eclipse.jetty.reactive.client.internal.AbstractSinglePublisher.request(AbstractSinglePublisher.java:90)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.subscriptions.SubscriptionHelper.deferredRequest(SubscriptionHelper.java:219)
	at io.reactivex.internal.subscribers.StrictSubscriber.request(StrictSubscriber.java:70)
```

Please, take a look. I'm not sure how to reproduce this error now, attach the code if I find the way.

**Comment by sbordet** [comment]
We need more information about this.
Can you share a program that reproduces the error, even randomly?
Are there specific websites that produce the error?
How can we reproduce the issue?

**Comment by sbordet** [comment]
FTR I tried to write a "slow" download test case, but it does not fail.

**Comment by sbordet** [comment]
We found the possible cause of this issue and a fix in #336.

**Comment by sbordet** [comment]
Fixed by #336.
